include/envoy/server/configuration.h

```diff
@@ -64,6 +64,13 @@ class Main {
    *         multiple nonresponsive threads.
    */
   virtual std::chrono::milliseconds wdMultiKillTimeout() const PURE;
+
+  /**
+   * @return double the percentage of threads that needs to meet the MultiKillTimeout before we
+   *         kill the process. If it is zero, then it'll fallback to the default behavior of
+   *         killing the process if two threads hit the multikill timeout.
```

Comment:
comment nit: The number of threads required for multi-kill is always at least 2.  This behavior applies beyond just the 0 default case.  You may want to refer to the max(2, registered_threads * percentage) computation.

Reply:
Tried to clarify it a bit more.